### PR TITLE
Proper spawning & Aspect Ratio stuff

### PR DESCRIPTION
**Pull request by devonpmack**
Spawn in empty positions, server stores position not relative to screen size and game scales it up.

note: i messed up collision a bit, we need to change the circle to scale to screen size.

### CODE REVIEW

File: server/main.js [hunk 1]
```diff
@@ -56,11 +56,18 @@ const public = path.join(__dirname, "..", "public");
 app.use(express.static(public));
 
 io.on("connection", function(socket) {
-  let x = 0;
-  let y = 0;
+  let x,y;
+
   getSpawn(map, x, y);
```
Comment (devonpmack) [inline server/main.js]: can we take this function out? What does it do?

File: server/main.js [hunk 2]
```diff
@@ -56,11 +56,18 @@ const public = path.join(__dirname, "..", "public");
...
   socket.emit("map", map);
-  console.log(map);
+
+  while (!x || map[x][y] === 1) {
+    x = Math.floor(Math.random() * height);
+    y = Math.floor(Math.random() * width);
+  }
+
+  x = x + 0.5;
+  y = y + 0.5;
```
Comment (devonpmack) [inline server/main.js:69]: keep picking random positions until we find an empty one

File: server/main.js [hunk 3]
```diff
@@ -70,6 +77,8 @@ io.on("connection", function(socket) {
     dy: 0
   });
 
+  socket.emit("startingPos", {x, y});
```
Comment (devonpmack) [inline server/main.js:79]: send the starting positiono

File: src/Game.js [hunk 4]
```diff
@@ -106,15 +109,15 @@ export default class Game {
             Math.sqrt(serverDiffX ** 2 + serverDiffY ** 2) >
             allowedServerDivergencePx
           ) {
-            playerObj.x = player.x;
-            playerObj.y = player.y;
+            playerObj.x = player.x * blockWidth;
+            playerObj.y = player.y * blockHeight;
```
Comment (devonpmack) [inline src/Game.js:113]: X will be from 0 - 16 so we multiply it by blockWidth to find the actual position

File: src/Player.js [hunk 5]
```diff
@@ -13,6 +18,13 @@ export default class Player {
     this.shot = {x1: 0, x2: 100, y1: 0, y2: 100, alpha: 0};
     this.hasShot = true;
 
+    if (socket) {
+      socket.on('startingPos', ({x,y}) => {
+        this.x = x * blockWidth;
+        this.y = y * blockHeight;
+      });
+    }
+
```
Comment (devonpmack) [inline src/Player.js:27]: in theory if they move before they spawn it might be possible to start in the 0-0 position

File: src/Player.js [hunk 6]
```diff
@@ -52,29 +64,20 @@ export default class Player {
   doOtherMovement(p) {
```
Comment (devonpmack) [inline src/Player.js:64]: we should combine the two movement functions into one but im saving that for later

File: src/Player.js [hunk 7]
```diff
@@ -1,9 +1,14 @@
 import io from "socket.io-client";
 
-const SPEED = 5;
+const blockHeight = window.innerHeight / 9; 
+const blockWidth = window.innerWidth / 16;
```
Comment (devonpmack) [inline src/Player.js:4]: duplicated information, but it should be fine for now

File: server/main.js (hunk 1 above)
Comment (vincendiary) [reply server/main.js]: This function was used to find an empty spot on the map grid, then move it into the x/y variables. If it's not used, we can remove it.

File: src/Player.js (hunk 6 above)
Comment (vincendiary) [reply src/Player.js:64]: Smooth and not smooth? I agree

File: src/Player.js (hunk 7 above)
Comment (vincendiary) [reply src/Player.js:4]: This was what I needed about 20 hours ago

File: src/Player.js (hunk 5 above)
Comment (vincendiary) [reply src/Player.js:27]: 0-0 should be fine if it's not a wall block.

**Review by vincendiary — COMMENTED** [review]
Changes look good so far. I'll add features when I get the chance.

File: src/Player.js (hunk 7 above)
Comment (devonpmack) [reply src/Player.js:4]: lol

File: server/main.js (hunk 1 above)
Comment (devonpmack) [reply server/main.js]: oh i see

Comment (devonpmack) [reply server/main.js]: removing...

File: src/Player.js (hunk 6 above)
Comment (devonpmack) [reply src/Player.js:64]: doing this later :)

**Comment by devonpmack** [comment]
Merging

**Pull request merged**